src/geom/SpanSet.cc

```diff
@@ -0,0 +1,640 @@
+                return yval < pr.y;
+            }
+};
+
+class ComparePrimaryRunM{
+ public:
+            bool operator()(PrimaryRun const& pr, int mval) {
+                return pr.m < mval;
+            }
+            bool operator()(int mval, PrimaryRun const& pr) {
+                return mval < pr.m;
+            }
+};
+}  // namespace
+
+namespace geom = lsst::afw::geom;
+
+// Expose properties of the underlying vector constaining spans such that the
+// SpanSet can be concidered a container
+geom::SpanSet::const_iterator geom::SpanSet::begin() const {
+    // Return the constant version as SpanSets should be immutable
+    return spanVector.cbegin();
+}
+
+geom::SpanSet::const_iterator geom::SpanSet::end() const {
+    // Return the constant version as SpanSets should be immutable
+    return spanVector.cend();
+}
+
+geom::SpanSet::const_iterator geom::SpanSet::cbegin() const {
+    return spanVector.cbegin();
+}
+
+geom::SpanSet::const_iterator geom::SpanSet::cend() const {
+    return spanVector.cend();
+}
+
+geom::SpanSet::const_reference geom::SpanSet::front() const {
+    // Return constant version as SpanSets should be immutable
+    return const_cast<geom::Span &>(spanVector.front());
+}
+
+geom::SpanSet::const_reference geom::SpanSet::back() const {
+    // Return constant version as SpanSets should be immutable
+    return const_cast<geom::Span &>(spanVector.back());
+}
+
+geom::SpanSet::size_type geom::SpanSet::size() const {
+    return spanVector.size();
+}
+
+bool geom::SpanSet::empty() const {
+    return spanVector.empty();
+}
+
+// Default constructor, creates a null null spanset which may be useful for
+// compairisons
+geom::SpanSet::SpanSet():spanVector(), bBox(), area(0) {}
+
+// Construct a spanset from a Box2I object
+geom::SpanSet::SpanSet(geom::Box2I const & box):bBox(box), area(box.getArea()) {
+    int beginY = box.getMinY();
+    int endY = box.getMaxY();
+
+    int beginX = box.getMinX();
+    int endX = box.getMaxX();
+
+    for (int i = beginY; i < (endY+1); ++i) {
+        spanVector.push_back(Span(i, beginX, endX));
+    }
+}
+
+// Construct a SpanSet from a std vector by copying
+geom::SpanSet::SpanSet(const std::vector<Span> & vec, bool normalize): spanVector(vec) {
+    if (normalize) {
+        runNormalize();
+    }
+    initialize();
+}
+
+// Construct a SpanSet from a std vector by moving
+geom::SpanSet::SpanSet(std::vector<Span> && vec, bool normalize): spanVector(std::move(vec)) {
+    if (normalize) {
+        runNormalize();
+    }
+    initialize();
+}
+
+void geom::SpanSet::runNormalize() {
+    // Ensure the span set is sorted according to Span < operator
+    std::sort(spanVector.begin(), spanVector.end());
+
+    // Now that the span is sorted, overlapping spans should be combined
+    // Start from 1 as the comparison will always be with the previous element
+    for (auto iter = ++(spanVector.begin()); iter != spanVector.end(); iter++) {
```

Comment:
It probably isn't a context where it matters, but the `++` form has the potential to be more efficient.